use-cases/vehicle-routing-time-windows/src/main/java/org/acme/vehiclerouting/domain/Depot.java

```diff
@@ -0,0 +1,28 @@
+package org.acme.vehiclerouting.domain;
+
+import com.fasterxml.jackson.annotation.JsonIdentityInfo;
+import com.fasterxml.jackson.annotation.JsonIdentityReference;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.ObjectIdGenerators;
+
+@JsonIdentityInfo(scope = Depot.class, generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
+public class Depot {
+
+    private final long id;
+
+    @JsonIdentityReference
+    private final Location location;
```

Comment:
Does @JsonIdentityReference do anything? The Location is serialized as a json array

Reply:
The same answer as below; the code needs polishing.